src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -21,15 +22,18 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
     /// </remarks>
     [Export(typeof(IThreadingContext))]
     [Shared]
-    internal sealed class ThreadingContext : IThreadingContext
+    internal sealed class ThreadingContext : IThreadingContext, IDisposable
     {
+        private readonly CancellationTokenSource _disposalTokenSource = new CancellationTokenSource();
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public ThreadingContext(JoinableTaskContext joinableTaskContext)
         {
             HasMainThread = joinableTaskContext.MainThread.IsAlive;
             JoinableTaskContext = joinableTaskContext;
             JoinableTaskFactory = joinableTaskContext.Factory;
```

Comment:
I don't see any code change here to *populate* the collection you're adding. Per the pattern, you're expected to create a new factory, passing the collection in as an argument. That would cause all JoinableTask's spawned by this JTF to automatically be added to the collection. 
Otherwise (as you have it now), any async task will have to be manually added to the collection, which might be easy to forget. But it gives you more control over which tasks will block, which maybe you need. But shouldn't *all* tasks block shutdown?

Reply:
I believe the goal is to make this change, but it's going to require a deeper review to ensure it doesn't cause problems.